yaksh/views.py

```diff
@@ -494,20 +499,29 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None):
             # validation
             if 'assignment' in request.FILES:
                 assignment_filename = request.FILES.getlist('assignment')
+                qp = QuestionPaper.objects.get(id=questionpaper_id)
             for fname in assignment_filename:
-                if AssignmentUpload.objects.filter(
-                    assignmentQuestion=current_question,
-                    assignmentFile__icontains=fname, user=user).exists():
-                    assign_file = AssignmentUpload.objects.get(
-                    assignmentQuestion=current_question,
-                    assignmentFile__icontains=fname, user=user)
+                assignment_files = AssignmentUpload.objects.filter(
+                            assignmentQuestion=current_question,
+                            assignmentFile__icontains=fname, user=user,
+                            question_paper=questionpaper_id)
+                if assignment_files.exists():
+                    assign_file = assignment_files.get(
+                            assignmentQuestion=current_question,
+                            assignmentFile__icontains=fname, user=user,
+                            question_paper=questionpaper_id)
                     os.remove(assign_file.assignmentFile.path)
                     assign_file.delete()
                 AssignmentUpload.objects.create(user=user,
-                    assignmentQuestion=current_question, assignmentFile=fname
+                    assignmentQuestion=current_question, assignmentFile=fname,
+                    question_paper=qp
                     )
             user_answer = 'ASSIGNMENT UPLOADED'
             if not current_question.grade_assignment_upload:
+                new_answer = Answer(question=current_question, answer=user_answer,
```

Comment:
there should be a label that states whether assignment is correct or wrong.
- This works similar to code question, marks are given to the assignment question if evaluated with hook else it does not give marks.

all the attempts for a given upload questions should be shown like we do for code questions, currently only the last one is shown.
- It is not done because we replace files if it is already submitted.

also every time user answer shown is 'ASSIGNMENT UPLOADED', it should be the answer from the file.
- In this case I feel it is not possible everytime to read the user uploaded file, since uploaded file can be of any type instead it would be better to give download option.

so every time you are replacing the assignment file for a given question, that is fine, but at least in database there should be a record of what was submitted via file.
And same you can display on UI, this will make the UI very useful.
- If you are telling to read student uploaded file again I don't think that it will be possible for every case. So I felt giving download option is better.